packages/effect/test/Stream/splitting.test.ts

```diff
@@ -90,6 +90,17 @@ describe("Stream", () => {
       assert.deepStrictEqual(Array.from(result), [])
     }))
 
+  it.effect("split - should infer proper types from refinement", () =>
+    Effect.gen(function*($) {
+      const stream = Stream.make(1, "2", 3)
+      const result = yield* $(
+        stream,
+        Stream.split((a: number | string): a is string => typeof a === "string"),
+        Stream.runCollect<Chunk.Chunk<number>, never, never>
+      )
+      assert.deepStrictEqual(Array.from(result).map((chunk) => Array.from(chunk)), [[1], [3]])
+    }))
+
```

Comment:
this is not necessary, for type level tests there's a `Stream.ts` file in the `dtslint` folder https://github.com/Effect-TS/effect/blob/main/packages/effect/dtslint/Stream.ts#L179-L193 (which is the cause of the CI failure btw)

p.s.
oh the title of the test should be amended btw "splitWhere" -> "split"